### PR TITLE
Options to move tmpdir, error_log, and pid file

### PR DESCRIPTION
**Pull request by geleman**
Allows for options to move other directories other than data_dir.
Ignore the init script because @daften is addressing that in #341

### CODE REVIEW

File: libraries/helpers.rb [hunk 1]
```diff
@@ -148,11 +145,12 @@ def init_records_script
...
 
         cat > /tmp/#{mysql_name}/my.sql <<-EOSQL
-DELETE FROM mysql.user ;
-CREATE USER 'root'@'%' IDENTIFIED BY '#{Shellwords.escape(new_resource.initial_root_password)}' ;
-GRANT ALL ON *.* TO 'root'@'%' WITH GRANT OPTION ;
+UPDATE mysql.user SET Password=PASSWORD('#{Shellwords.escape(new_resource.initial_root_password)}');
+DELETE FROM mysql.user WHERE user LIKE '';
+DELETE FROM mysql.user WHERE user = 'root' and host NOT IN ('127.0.0.1','localhost');
+DROP DATABASE test;
+DELETE FROM mysql.db WHERE db LIKE 'test%'
```
Comment (daften) [inline libraries/helpers.rb]: You update the init script here, that is #341, so shouldn't be in here I think

**Comment by daften** [comment]
@geleman This probably makes sense, but for a non dbadmin like me, could you explain the use cases or give an example when moving the pidfile or tmp dir would be needed? :)

**Comment by geleman** [comment]
For the pidfile and the socket I think you should have the option to customize it yourself or leave it as default. My current org always buried it deep in it's directory structure. I like putting it in /tmp myself and there are plenty others out there that I feel would feel the same way. 

As for the tmpdir it should not be on disk. Which by putting it in /tmp it is hard coded to have any tmp table that is created by MySQL to go to disk. You should put it in memory for maximum performance. It makes a huge difference in that aspect by having it in memory. My current set up is /tmp/shm becasue I get rid of /dev/shm and mount /tmp/shm as a tmpfs file system with rw,noatime options.

File: libraries/helpers.rb (hunk 1 above)
Comment (geleman) [inline libraries/helpers.rb]: I'm still trying to figure out how to use git as well. I wasn't sure how to exclude showing that change

Comment (daften) [inline libraries/helpers.rb]: Oh, ok, I don't know on the command-line, but with other tools e.g. sourcetree, you can do that fairly easily. With the client tool from github itself you can also do that I think :)

**Comment by daften** [comment]
Oh, ok, those are some heavy optimizations I didn't even know were possible (for the tmpdir). Thanks for the input! :)

**Comment by someara** [comment]
Hi.

I slacked on merging this and now there are conflicts... would you mind rebasing this for me?

**Comment by geleman** [comment]
Could you tell me what you mean by rebasing? I hope you can get it merged
my organization is going to base all future installs with this and the
root@localhost change.

**Comment by someara** [comment]
I mean...

git checkout master
git pull origin master
git checkout <your-branch-here>
git rebase master

*fix conflicts*

re-test

git push <your_remote> <your_branch>

=)

**Comment by geleman** [comment]
I will get it worked out. Not only am I a chef rookie but a git rookie
also. MySQL has been my forte and I'm adding these tools to my toolbox. So
if you can please bare with me a little.

**Comment by geleman** [comment]
I am having issues with rebasing. May need to just do a new fork and make changes again. We are still trying but this may be the route we have to take. Sorry about my lack of skill in this area.

**Comment by someara** [comment]
No worries!

-s

**Comment by daften** [comment]
@geleman This resource could be of help: https://git-scm.com/book/en/v2/Git-Tools-Rewriting-History
Otherwise, feel free to contact me on skype (same username as here) if you need assistance, maybe I can help :)

**Comment by geleman** [comment]
Okay I have reforked and added my changes and it ran fine, but I am getting issues with the mysql_config provider on the restart. I didn't change anything with that and it worked prior. Were there some changes with that provider that I am not seeing?

**Comment by geleman** [comment]
I have confirmed my changes work as evidence by the default my.cnf file:

[root@mysql1 vagrant]# cd /etc/mysql-test/
[root@mysql1 mysql-test]# ll
total 8
drwxr-x---. 2 mysql mysql 4096 Jul 13 15:57 conf.d
-rw-------. 1 mysql mysql  700 Jul 13 15:57 my.cnf
[root@mysql1 mysql-test]# cat my.cnf
# Chef generated my.cnf for instance mysql-test

[client]
default-character-set          = utf8
port                           = 3306
socket                         = /tmp/mysqld.sock

[mysql]
default-character-set          = utf8

[mysqld]
user                           = mysql
pid-file                       = /tmp/mysqld.pid
socket                         = /tmp/mysqld.sock
bind-address                   = 0.0.0.0
port                           = 3306
datadir                        = /data/mysql
tmpdir                         = /tmp/shm
log-error                      = /logs/mysql/mysql.err

You can see that my log-error, tmpdir, pid are all located based on my changes. 
However the issue I keep having arises when I run the mysql_config provider with my attirbutes file I have created. 

I use the attribute file for mysql_service as well. This is my mysql_service comand recipie

mysql_service node['mysql_config']['instance_name'] do
  port '3306'
  version '5.6'
  data_dir '/data/mysql'
  tmp_dir '/tmp/shm'
  error_log '/logs/mysql/mysql.err'
  socket '/tmp/mysqld.sock'
  pid_file '/tmp/mysqld.pid'
  bind_address '0.0.0.0'
  initial_root_password opts['password']
  action [:create, :start]
end

So as stated this runs fine without the mysql_config provider.
Once I add that:

mysql_config node['mysql_config']['instance_name'] do
  instance node['mysql_config']['instance_name']
  source 'defaults.cnf.erb'
  action :create
  notifies :restart, "mysql_service[#{node['mysql_config']['instance_name']}]"

It fails.
This is the same exact settings I ran prior to forking the latest version 6.0.31.

I'm ready to update the pull request with my changes but this is preventing me from doing so even though it has nothing to do with the changes i have made.

**Comment by geleman** [comment]
I must also add that if I change mysql_service to something static and also reflect those changes in mysql_config it still fails. So it seems there is an issue happening with attributes with the mysql_config provider.

**Comment by daften** [comment]
That's most likely because of chef 12.4, Try using 12.3.0 (by using vagrant-omnibus)

**Comment by someara** [comment]
12.4 should be fixed up btw

**Comment by geleman** [comment]
I'm using the latest in my vagrant which is 12.4.1 and this is failing. If I go to 12.3 then it's all for not.

**Comment by someara** [comment]
Maybe the contents of your defaults.cnf.erb file are causing a conflict that causes the service not to start?

**Comment by geleman** [comment]
No I checked and it's fine. I'm even check the log file for errors. Nothing
was changed in the default.cnf.erb

**Comment by geleman** [comment]
Only other option is to remove attributes and test and then put in
incrementally

**Comment by someara** [comment]
Just send your PR and I'll make sure its kosher =)
-s

**Comment by geleman** [comment]
I will first thing in the am.

**Comment by someara** [comment]
closing this pull for now, since you removed your branch.

**Comment by geleman** [comment]
Should just do another or reopen?

**Comment by someara** [comment]
Easiest thing to do is send another PR

**Pull request closed without merging**